core/deployment/src/main/java/io/quarkus/deployment/recording/RecorderContext.java

```diff
@@ -44,12 +44,12 @@ <F, T> void registerSubstitution(Class<F> from, Class<T> to,
      * Creates a Class instance that can be passed to a recording proxy as a substitute for a class that is not loadable
      * at processing time. At runtime the actual class will be passed into the invoked method.
      *
-     * @param name The class name
+     * @param name The fully-qualified class name
      * @return A Class instance that can be passed to a recording proxy
-     * @deprecated This construct is no longer needed since directly loading deployment/application classes at
-     *             processing time in build steps is now safe
+     * @deprecated This construct should not be needed in most use cases since directly loading deployment/application classes
+     *             at processing time in build steps is safe. However, there are use cases where this method comes in handy.
```

Comment:
When I read the last part of the note, the first question I have in mind is: which use cases? IMHO I would either remove this part or give at least one use case where it makes sense otherwise it feels like it is an incomplete sentence, don't you agree?

Reply:
I do but I'm not sure we could describe a simple use case in one sentence. @Ladicek do you have an idea?

Reply:
I don't think this is necessary, but the prime use case I have in mind is classes generated by earlier build steps. How about this?

> This construct should not be needed in most cases since directly loading deployment/application classes
> in build steps is safe. However, there are cases where this method comes in handy, such as referring to
> classes that were generated in previous build steps (see {@link GeneratedClassBuildItem}).

Reply:
Perfect, I will update both the javadoc and the docs...